posthog/cdp/templates/avo/test_template_avo.py

```diff
@@ -138,3 +140,71 @@ def test_property_filters(self):
 
             res = self.get_mock_fetch_calls()[0][1]["body"][0]["eventProperties"]
             assert [item["propertyName"] for item in res] == expected_result
+
+
+class TestTemplateMigration(BaseTest):
+    def get_plugin_config(self, config: dict):
+        _config = {
+            "avoApiKey": "1234567890",
+            "environment": "dev",
+            "appName": "PostHog",
+        }
+        _config.update(config)
+        return PluginConfig(enabled=True, order=0, config=_config)
+
+    def test_default_config(self):
+        obj = self.get_plugin_config({})
+        template = TemplateAvoMigrator.migrate(obj)
+        assert template["inputs"] == snapshot(
+            {
+                "apiKey": {"value": "1234567890"},
+                "environment": {"value": "dev"},
+                "appName": {"value": "PostHog"},
+                "excludeProperties": {"value": ""},
+                "includeProperties": {"value": ""},
+            }
+        )
+        assert template["filters"] == {"events": []}
+
+    def test_include_events(self):
+        obj = self.get_plugin_config({"includeEvents": "sign up,page view"})
+        template = TemplateAvoMigrator.migrate(obj)
+        assert template["inputs"] == snapshot(
+            {
+                "apiKey": {"value": "1234567890"},
+                "environment": {"value": "dev"},
+                "appName": {"value": "PostHog"},
+                "excludeProperties": {"value": ""},
+                "includeProperties": {"value": ""},
+            }
+        )
+        assert template["filters"] == {
+            "events": [
+                {"id": "sign up", "name": "sign up", "type": "events", "order": 0},
+                {"id": "page view", "name": "page view", "type": "events", "order": 0},
+            ]
+        }
+
+    def test_exclude_events(self):
+        obj = self.get_plugin_config({"excludeEvents": "sign up,page view"})
+        template = TemplateAvoMigrator.migrate(obj)
+        assert template["inputs"] == snapshot(
+            {
+                "apiKey": {"value": "1234567890"},
+                "environment": {"value": "dev"},
+                "appName": {"value": "PostHog"},
+                "excludeProperties": {"value": ""},
+                "includeProperties": {"value": ""},
```

Comment:
Would've been good to test these two getting a config as well, but we can live without.

Reply:
added

Reply:
added